controllers/validators/avatar.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const validate = require('./validate-by-schema');
+
+const typeOf = require('image-type');
+const fs = require('fs-extra');
+const path = require('path');
+
+/*
+ * Patching avatar
+ */
+const patchHeaders = validate('patchAvatarHeaders');
+const patchFile = validate('patchAvatarFile');
+
+/**
+ * Express middleware to check whether image has a supported mime-type
+ */
+const patchFileType = async function (req, res, next) {
+  const supportedMimes = ['image/png', 'image/jpeg'];
+
+  // validate file type
+  const filePath = path.resolve(`./${req.file.path}`);
+  // - read the image
+  const fileBuffer = await fs.readFile(filePath);
```

Comment:
.

Reply:
Nice. This can actually happen. We wipe the temporary uploads folder every day, not gracefully. (there is a TODO keep files younger than 1 minute). Error will be handled.